### PR TITLE
Add method for ParamterSweepTool to change lower/upper bounds, and fix/unfix variables. 

### PR DESCRIPTION
**Pull request by avdudchenko**
## Summary/Motivation:

This adds new optional arguments to parameter sweep options,enabling user to sweep across lower bounds, upper bound, or fix and unfix variables. Primary objective is to enable complete depreciation of prior method of passing in a model directly into a PS tool and instead requiring user to pass in build_model function. 

This future change will mean a user would need to explicitly specify lower/upper bound or fixed/unfixed states in model init or optimize function.

This PR enables user to now include the changed lower/upper bound or fixed/unfixed states as part of sweep_paramters, ensuring these values are updated after a model is built or initialized.  

This update is also meant for the WaterTAP UI which needs to update the mode with user supplied variables before running a sweep, currently UI relies on old ParamterSweep interface, this will enable us to migrate to new interface and enable local parallel utilization. 

## Changes proposed in this PR:
-add SetMode class, the defines if PS tool should fix variable, set upper or lower bound, or fix unfix variables
-Includes updates to ParameterSweepReader and tests through out.

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: watertap/tools/parameter_sweep/parameter_sweep.py [hunk 1]
```diff
@@ -722,7 +775,7 @@ def parameter_sweep(self, *args, **kwargs):
         pass
 
 
-class ParameterSweep(_ParameterSweepBase, _ParameterSweepParallelUtils):
+class ParameterSweep(_ParameterSweepBase, _ParameterSweepParallelUtils):  #
```
Comment (k1nshuk) [inline watertap/tools/parameter_sweep/parameter_sweep.py]:
```suggestion
class ParameterSweep(_ParameterSweepBase, _ParameterSweepParallelUtils):
```

**Review by bknueven — APPROVED** [review]

**Review by k1nshuk — APPROVED** [review]
Can be merged in after adding the docs.

**Review by k1nshuk — APPROVED** [review]
Looks good to me. Thanks for adding the documentation.

**Comment by avdudchenko** [comment]
This is ready for merge.

**Pull request merged**